pkg/logging/factory.go

```diff
@@ -0,0 +1,79 @@
+package logging
+
+import (
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"os"
+	"time"
+)
+
+//
+// Builder.
+type Builder interface {
+	New() logr.Logger
+	V(int, logr.Logger) logr.Logger
+}
+
+//
+// Zap builder factory.
+type ZapBuilder struct {
+}
+
+//
+// Build new logger.
+func (b *ZapBuilder) New() (l logr.Logger) {
+	var encoder zapcore.Encoder
+	var options []zap.Option
+	sinker := zapcore.AddSync(os.Stderr)
+	if Settings.Development {
+		cfg := zap.NewDevelopmentEncoderConfig()
+		encoder = zapcore.NewConsoleEncoder(cfg)
+		options = append(
+			options,
+			zap.Development(),
+			zap.AddStacktrace(zap.ErrorLevel))
+	} else {
+		cfg := zap.NewProductionEncoderConfig()
+		encoder = zapcore.NewJSONEncoder(cfg)
+		options = append(
+			options,
+			zap.AddStacktrace(zap.WarnLevel),
+			zap.WrapCore(
+				func(core zapcore.Core) zapcore.Core {
+					return zapcore.NewSampler(
+						core,
+						time.Second,
+						100,
+						100)
+				}))
+	}
+	level := zap.NewAtomicLevelAt(zap.DebugLevel)
```

Comment:
Is a new logger supposed to always default to Debug level regardless of development mode?

Reply:
Yes.  This ensure that it will log everything and _our_ logger can controller verbosity.